zephyr/CMakeLists.txt

```diff
@@ -25,8 +31,29 @@ if(CONFIG_WOLFSSL)
     zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/src/tls.c)
     zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/src/tls13.c)
     zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/src/wolfio.c)
-  
+
+    # FIPS Boundary
+    if(CONFIG_WOLFCRYPT_FIPS)
+    zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/wolfcrypt/src/wolfcrypt_first.c)
+    endif()
+    zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/wolfcrypt/src/hmac.c)
```

Comment:
Since you're doing a major refactor, I think it may be easier to use wildcards for source files. You can see an example of how I did this in wolfMQTT https://github.com/wolfSSL/wolfMQTT/blob/master/zephyr/CMakeLists.txt.

Reply:
That won't allow definition of the FIPS boundary.